src/XMakeBuildEngine/Construction/ProjectElementContainer.cs

```diff
@@ -540,6 +545,15 @@ internal void AddToXml(ProjectElement child)
             }
         }
 
+        private static bool HasXmlRepresentation(ProjectElement element)
```

Comment:
It wasn't enough in the original PR because I only put `Implicit` on `ProjectImportElement`. But you're right: if I just extend it to all `ProjectElement`s this becomes much cleaner.